.chloggen/chore_25128_regex.yaml

```
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: k8sattributeprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Deprecate FieldExtractConfig.Regex parameter in favor of `extract()` funrction in metricstransform processor
```

```suggestion
note: Deprecate FieldExtractConfig.Regex parameter in favor of `extract()` function in the transform processor
```
We also need a note about what release this configuration option will be removed. In the issue it was proposed to remove it 5 releases after being deprecated, so maybe make a note that it will be removed in `v0.108.0`?